core/core-security/src/main/java/com/pgms/coresecurity/security/service/OAuth2AuthenticationSuccessHandler.java

```diff
@@ -0,0 +1,71 @@
+package com.pgms.coresecurity.security.service;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.UUID;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.oauth2.core.user.DefaultOAuth2User;
+import org.springframework.security.web.authentication.AuthenticationSuccessHandler;
+import org.springframework.stereotype.Component;
+
+import com.pgms.coredomain.domain.member.Member;
+import com.pgms.coredomain.domain.member.Role;
+import com.pgms.coredomain.domain.member.enums.Provider;
+import com.pgms.coredomain.domain.member.repository.MemberRepository;
+import com.pgms.coredomain.domain.member.repository.RoleRepository;
+import com.pgms.coresecurity.security.jwt.JwtUtils;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class OAuth2AuthenticationSuccessHandler implements AuthenticationSuccessHandler {
+
+	private final MemberRepository memberRepository;
+	private final RoleRepository roleRepository;
+	private final JwtUtils jwtUtils;
+
+	@Override
+	@Transactional
+	public void onAuthenticationSuccess(HttpServletRequest request, HttpServletResponse response,
+		Authentication authentication) throws IOException {
+		DefaultOAuth2User defaultOAuth2User = (DefaultOAuth2User)authentication.getPrincipal();
+
+		Map<String, Object> properties = defaultOAuth2User.getAttribute("properties");
+		String name = (String)properties.get("nickname");
+
+		Map<String, Object> kakaoAccount = defaultOAuth2User.getAttribute("kakao_account");
+		String email = (String)kakaoAccount.get("email");
+
+		final Member member = (Member)memberRepository.findByEmailWithRole(email).orElseGet(() -> {
```

Comment:
```suggestion
		final Member member = memberRepository.findByEmailWithRole(email).orElseGet(() -> {
```
repository에서 타입을 명시하면 타입 캐스팅을 안해줘도 될것 같습니당!!